filesystem/client_interface.go

```diff
@@ -0,0 +1,11 @@
+package filesystem
+
+import "io"
+
+type Client interface {
+	OpenReadCloser(name string) (io.ReadCloser, error)
+	OpenWriteCloser(name string) (io.WriteCloser, error)
+	Exists(name string) (bool, error)
+	Rename(oldpath, newpath string) error
+	Glob(pattern string) (matches []string, err error)
```

Comment:
@BaiGang @xiaoyunwu 
Glob has been fixed according to your needs.
It currently supports "*" and "?" matching.